### DISCUSSION TITLE
Fetcher calls only after variable delay, but only on prod builds in a Next.js application

### DISCUSSION THREAD

**Discussion by kylemh** [vercel/swr#1090] [Q&A] [answered]
I've never seen an error this weird, and have no idea how to get around it.

I've got the following hook:

```tsx
import useSWR from 'swr';
import { Account as AccountAPI } from '@air/api';
import { Account as TAccount } from '@air/api/types';
import { ACCOUNT } from '@constants/swrKeys';

export const useAccount = () => {
  const { data: account, error, mutate } = useSWR<TAccount | null>(ACCOUNT, AccountAPI.get);

  return {
    account,
    error,
    isLoading: !account && !error,
    key: ACCOUNT,
    mutateAccount: mutate,
  };
};
```

Used in a page file like so: 

```tsx
export default function SomePage() {
  const { account } = useAccount();
  return <></>
}
```

The fetch under the hood of `AccountAPI.get()` doesn't actually fire until 5-10s after the page loads. I have other fetches that aren't within SWR hooks that are evaluating instantly. I also did the following and the promise evaluated instantly as-expected.

```tsx
export default function SomePage() {
  const [account, setAccount] = useState<TAccount | null>(null);

  useEffect(() => {
    AccountAPI.get().then(_account => setAccount(_account));
  }, []);

  return <></>
}
```

---

**What makes this even more baffling is that the SWR hook is evaluating instantly when using the dev server, but not when leveraging prod.**

### Running a prod build locally (see 5s delay)
![with-delay](https://user-images.githubusercontent.com/9523719/113380663-ac665400-9331-11eb-8423-303d82856f1a.gif)

### Running with dev server (no delay)
![no-delay](https://user-images.githubusercontent.com/9523719/113380684-b8521600-9331-11eb-9ccc-ac8374aec8c0.gif)

**Comment by shuding** [comment]
This is indeed very weird, is it possible to create a minimal reproduction on https://codesandbox.io (it's not a production build, though)? I cannot reproduce this locally. Can you confirm that the `useAccount()` call isn't delayed?

Also which SWR version was it? If you are on the latest, `swr@beta` might worth a try since there is no breaking changes.

**Reply by kylemh** [reply]
I was using 0.4.2, and got very hopeful and upgrade would solve my issue, but updated to latest and still saw the issue.

I confirmed at least with logging statements that the function containing the `useSWR` reference is being called immediately.

I can try to create a reproduction.

**Reply by kylemh** [reply]
Debugging a bit further before I try to create a reproduction (cuz it'll be a bit of working finding an open API and putting it together)...

I tried using `onErrorRetry: (error) => { console.error({ error }); }` within the useSWR's options wondering i maybe some error was getting swallowed. I end up seeing a log statement that says `{ error: undefined }`... If the error is undefined, why is onErrorRetry being called?

Also, why would there be no related network call for this fail?

![Screen Shot 2021-04-02 at 2 09 51 PM](https://user-images.githubusercontent.com/9523719/113454464-2132a000-93bd-11eb-8bd5-d4a5d4509e96.png)

**Reply by huozhi** [reply]
Does your fetcher having cases that throw `undefined` as error? I'm suspecting that's where might trigger the try-catch capture it.

**Reply by kylemh** [reply]
Nope 😞 
![Screen Shot 2021-04-07 at 11 29 08 PM](https://user-images.githubusercontent.com/9523719/113978698-17fc6580-97f9-11eb-8acf-af6ad2a8f5f6.png)
![Screen Shot 2021-04-07 at 11 29 00 PM](https://user-images.githubusercontent.com/9523719/113978703-192d9280-97f9-11eb-9709-675dc4d9ad03.png)

The errors go straight into SWR, react-dom, bugsnag, next.js internals, and then webpack

**Comment by kylemh** [comment, marked as answer]
There is indeed an uncaught error within my fetcher 🤦🏽‍♂️
